client/me/purchases/index.js

```diff
@@ -58,11 +58,14 @@ export default function( router ) {
 		clientRender
 	);
 
+	/**
+	 * The siteSelection middleware has been removed from this root.
+	 * No selected site!
+	 */
 	router(
 		paths.managePurchase( ':site', ':purchaseId' ),
```

Comment:
If we no longer evaluate it, can we also drop the `:site` fragment from purchase routes?

Reply:
Ah, just noticed #25219. So maybe in a subsequent iteration?

(I'm pretty positive we should get rid of it if we don't use it, and if we actually don't want to use it on a conceptual level.)

Reply:
Someday, but trying to keep the scope of this focused for now.

The trick is that pretty much _all_ of the routes need it right now, at least to move between routes. That's complicated by the fact that most routes depend on the siteSelection route middleware which _does_ require siteSlug in the URL. I'd like to transition all purchases away from that, but that's a bigger issue.